### DISCUSSION TITLE
Docker automatically setup

### DISCUSSION THREAD

**Discussion by jannismilz** [pocketbase/pocketbase#1051] [Q&A] [answered]
I'm using https://github.com/muchobien/pocketbase-docker and I want to automatically create e.g. admin user or upload settings when I run the container. Is there a way you can implement that as well?

There would be a workaround but that's very annoying... I could just create a start script and would give them an admin user so it can update settings and collection structure via API call.

**Comment by ganigeorgiev** [comment, marked as answer]
Currently, in addition to the API call script, you can also:
- create a test pb_data/data.db or run custom sql import script against it
- create a [Go db migration](https://pocketbase.io/docs/db-migrations/) if you use PocketBase as framework (more documentation on this will be available soon)

After v0.8.0, I'll research allowing custom db migrations for the prebuilt binary too using a JS interpreter like goja or otto.

**Reply by jannismilz** [reply]
Can I post a db structure and everything without having set up an admin account? Like without even using the webdashboard?

**Reply by ganigeorgiev** [reply]
If you are using the API, you first need to submit a `POST /api/admins` to create an Admin account and then authenticate with it.

**Reply by jannismilz** [reply]
Okay thanks, so in this case I can create an admin account with the api itself at the beginning right?

**Reply by ganigeorgiev** [reply]
Yes, when there is no admin, the `POST /api/admins` action is "public" and you can send a request to it without authorizing.

**Reply by jannismilz** [reply]
Awesome, thanks!
